prober/utils.go

```diff
@@ -32,7 +32,7 @@ var protocolToGauge = map[string]float64{
 }
 
 // Returns the IP for the IPProtocol and lookup time.
-func chooseProtocol(ctx context.Context, IPProtocol string, fallbackIPProtocol bool, target string, registry *prometheus.Registry, logger log.Logger) (ip *net.IPAddr, lookupTime float64, returnerr error) {
+func chooseProtocol(ctx context.Context, IPProtocol string, fallbackIPProtocol bool, target string, registry *prometheus.Registry, logger log.Logger) (ip *net.IPAddr, lookupTime float64, err error) {
```

Comment:
How about we switch this to un-named returns?

Reply:
We set lookupTime in a defer, once for all:

https://github.com/prometheus/blackbox_exporter/pull/796/files#diff-b99c121217776765037cbc7e4dfca853b9ef8243cb2f42f4a95f92c7afbb0361L66-L68

Since we use this 3 or 4 times, I think that the named returns are a better solution that repeating this multiple times.

Reply:
ok